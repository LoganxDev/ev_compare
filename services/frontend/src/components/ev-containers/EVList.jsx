import React from 'react';

const EVList = (props) => {
    return (
        <div>
            {
                props.evs.map((ev) => {
                    return (
                        <h4
                            key={ev.id}
                            className="box title is-4"
                        >{ ev.brand }{ ev.make }
                        </h4>
                    )
                })
            }
        </div>
    )
};

export default EVList;