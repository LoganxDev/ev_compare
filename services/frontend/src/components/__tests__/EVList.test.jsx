import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import EVList from '../EVList';

const evs = [
    {
        'make': 'Model 3',
        'brand': 'Tesla',
        'id': 1,
        'storage': 15
    },
    {
        'make': 'Model S',
        'brand': 'Tesla',
        'id': 2,
        'storage': 32
    }
];

test('EVList renders properly', () => {
    const wrapper = shallow(<EVList evs={evs} />);
    const element = wrapper.find('h4');
    expect(element.length).toBe(2);
    expect(element.get(0).props.children[0]).toBe('Tesla');
});

test('EVList renders a snapshot correctly', () => {
    const tree = renderer.create(<EVList evs={evs} />).toJSON();
    expect(tree).toMatchSnapshot();
});