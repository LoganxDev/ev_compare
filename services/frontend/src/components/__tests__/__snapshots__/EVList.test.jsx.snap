// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EVList renders a snapshot correctly 1`] = `
<div>
  <h4
    className="box title is-4"
  >
    Tesla
    Model 3
  </h4>
  <h4
    className="box title is-4"
  >
    Tesla
    Model S
  </h4>
</div>
`;
